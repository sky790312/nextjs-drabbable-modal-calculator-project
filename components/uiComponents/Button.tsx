import React, { ReactNode } from 'react'
import styled from 'styled-components/'

interface Props extends StyledButtonProps {
  children: ReactNode
  onClick: () => void
}

type StyledButtonProps = {
  size?: 'md' | 'block'
  buttonStyle?: 'blue' | 'gray' | 'darkGray'
}

const defaultStyle: StyledButtonProps = {
  size: 'md',
  buttonStyle: 'gray'
}

export const Button: React.FC<Props> = React.memo(({ children, size = defaultStyle.size, buttonStyle = defaultStyle.buttonStyle, onClick }) => {
  return <StyledButton buttonStyle={buttonStyle} size={size} onClick={onClick}>{children}</StyledButton>;
});

const StyledButton = styled.button<StyledButtonProps>`
  border: none;
  outline: none;
  font-size: 24px;

  ${props => {
    switch (props.buttonStyle) {
      case 'gray':
        return `
          color: ${props.theme.colors.black};
          background-color: ${props.theme.colors.gray};
        `
      case 'blue':
        return `
          color: ${props.theme.colors.white};
          background-color: ${props.theme.colors.blue};
        `
      case 'darkGray':
        return `
          color: ${props.theme.colors.white};
          background-color: ${props.theme.colors.darkGray};
        `
    }
  }}

  ${props => {
    switch (props.size) {
      case 'md':
        return `
          width: 80px;
          height: 80px;
          border-radius: 50%;
        `
      case 'block':
        return `
          width: 100%;
          height: 80px;
          border-radius: 40px;
        `
    }
  }}
`