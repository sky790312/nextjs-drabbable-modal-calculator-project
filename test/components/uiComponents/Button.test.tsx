import React from 'react'
import { render, withProviders } from '@/test/testUtils'
import { Button } from '@/components/uiComponents/Button'

describe('Button component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(withProviders(<Button>button</Button>), {})
    expect(asFragment()).toMatchSnapshot()
  })
})
