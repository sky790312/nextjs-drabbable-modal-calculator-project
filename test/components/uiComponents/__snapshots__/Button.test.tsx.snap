// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ eCqVEX"
  >
    button
  </button>
</DocumentFragment>
`;
